openedx/core/djangoapps/video_config/utils.py

```diff
@@ -0,0 +1,92 @@
+"""
+Video configuration utilities
+"""
+
+import logging
+
+from django.conf import settings
+from opaque_keys.edx.keys import UsageKey
+
+from openedx.core.djangoapps.video_config.toggles import PUBLIC_VIDEO_SHARE
+from openedx.core.lib.courses import get_course_by_id
+
+log = logging.getLogger(__name__)
+
+# Video sharing constants
+COURSE_VIDEO_SHARING_PER_VIDEO = 'per-video'
+COURSE_VIDEO_SHARING_ALL_VIDEOS = 'all-on'
+COURSE_VIDEO_SHARING_NONE = 'all-off'
+
+
+class VideoSharingUtils:
+    """
+    Provides utility methods for video sharing functionality.
+    """
+
+    @staticmethod
+    def get_public_video_url(usage_id: UsageKey) -> str:
+        """
+        Returns the public video url
+        """
+        return fr'{settings.LMS_ROOT_URL}/videos/{str(usage_id)}'
+
+    @staticmethod
+    def is_public_sharing_enabled(video_block):
```

Comment:
by taking an instance of VideoBlock as a parameter, this function is making assumptions about the fields of VideoBlock, which it should not be once the VideoBlock is defined outside of the platform.

could this function take just `(usage_key: UsageKey, public_access: bool)` instead?